exploration/number-selection.md

```diff
@@ -92,45 +95,241 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
+  (This function is valid as a formatter)
+
+The function `:integer` operates identically to `:number`, except:
+- the `minimumFractionDigits` and `maximumFractionDigits` options
+  default to `0` for all values
+- the _options_ `minimumFractionDigits` and `maximumFractionDigits`
+  are invalid.
+
+### Operands
+
+The operand of a number selector is either a literal that matches the `number-literal`
+production in the [ABNF](/main/spec/message.abnf) and which is parsed by the 
+implementation into a number; or an implementation-defined numeric type; 
+or is some serialization of a number that the implementation recognizes.
+
+> For example, in Java, any subclass of `java.lang.Number` plus the primitive
+> types (`boolean`, `byte`, `short`, `int`, `long`, `float`, `double`, etc.) 
+> might be considered as the "implementation-defined numeric types".
+> Implementations in other programming languages would define different types
+> or classes according to their local needs.
+
+Implementations MUST treat literals that do not match `number-literal` as
+non-numeric string literals.
+Non-numeric values, including non-numeric string literals, result in
+a _Selection Error_.
+
+> _Ed note:_ Alternatively we can require that non-numeric values result in `*`
+
+### Options
+
+All number selectors have the following options internally.
+Unless otherwise specified in [functions](#functions) all of these options
+can be used in _annotations_.
+
+> [!IMPORTANT]
+> Some functions do not expose one or more of these options.
+> Some functions have different defaults.
+> See [functions](#functions) above.
+
+- `select`
+   -  `plural` (default)
+   -  `ordinal`
+   -  `exact`
+- `compactDisplay`
+   - `short` (default)
+   - `long`
+- `currency`
+   -  (`alpha3` ISO 4217 currency code)
+- `currencyDisplay`
+   - `symbol` (default)
+   - `narrowSymbol`
+   - `code`
+   - `name`
+- `currencySign`
+  - `accounting`
+  - `standard` (default)
+- `notation`
+   - `standard` (default)
+   - `scientific`
+   - `engineering`
+   -  `compact`
+- `numberingSystem`
+   -  (one of the ISO15927 script codes or extensions defined by ICU and/or Intl.NumberFormat
+      e.g. arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn 
+      limb mlym mong mymr orya tamldec telu thai tibt)
+- `signDisplay`
+   -  `auto` (default)
+   -  `always`
+   -  `exceptZero`
+   -  `never`
+- `style`
+  - `decimal` (default)
+  - `currency`
+  - `percent`
+  - `unit`
+- `unit`
+   - (anything not empty)
+- `unitDisplay`
+   - `long`
+   - `short` (default)
+   - `narrow`
+- `minimumIntegerDigits`
+  - (positive integer, default: `1`)
+- `minimumFractionDigits`
+  - (positive integer)
+- `maximumFractionDigits`
+  - (positive integer)
+- `minimumSignificantDigits`
+  - (positive integer)
+- `maximumSignificantDigits`
+  - (positive integer)
+
+
+> [!NOTE]
+> We have a requirement for `offset` but have not provided for it yet??
```

Comment:
Right, something like that would need to be added. Lacking it means that we can't emulate a major feature of MF1 behavior with MF2 messages, so that needs to be documented clearly in the spec for the tech preview, so that people don't try to do it.

Reply:
This could be supported by an option like `icu:offset`, much like skeletons, another MF1 feature that the initial core set of built-in functions won't support.

Reply:
Note: I think we would need to add icu:offset in the tech preview in ICU, because otherwise we wouldn't be able to emulate MF1.0 with MF2.0, which is an important test.

Reply:
@macchiati Note that we won't be able to emulate the formats that we agreed to exclude: `spellout`, `duration`, and `choice` unless ICU chooses to implement them using namespacing (`:icu:spellout`, etc.)